### PR TITLE
add synonym api

### PR DESCRIPTION
**Pull request by maxiloc**

### CODE REVIEW

**Comment by maxiloc** [comment]
Ready for review

**Comment by redox** [comment]
@maxiloc there are mismatches between `readHostsArray` and `writeHostsArray`

File: src/AlgoliaSearch/Index.php [hunk 1]
```diff
@@ -1081,6 +1081,143 @@ public function browseFrom($query, $params = null, $cursor = null)
...
+            '/1/indexes/'.$this->urlIndexName.'/synonyms/clear?forwardToSlaves='.($forwardToSlaves ? 'true' : 'false'),
+            null,
+            null,
+            $this->context->readHostsArray,
+            $this->context->connectTimeout,
+            $this->context->readTimeout
+        );
+    }
+
+    public function batchSynonyms($objects, $forwardToSlaves = false, $replaceExistingSynonyms = false)
```
Comment (redox) [inline src/AlgoliaSearch/Index.php]: No comment/doc on this one?

File: src/AlgoliaSearch/Index.php [hunk 2]
```diff
@@ -1081,6 +1081,143 @@ public function browseFrom($query, $params = null, $cursor = null)
...
+                .'&forwardToSlaves='.($forwardToSlaves ? 'true' : 'false'),
+            null,
+            $objects,
+            $this->context->readHostsArray,
+            $this->context->connectTimeout,
+            $this->context->readTimeout
+        );
+    }
+
+    public function saveSynonym($objectID, $content, $forwardToSlaves = false)
```
Comment (redox) [inline src/AlgoliaSearch/Index.php]: No comment/doc on this one?

File: src/AlgoliaSearch/Index.php [hunk 3]
```diff
@@ -1081,6 +1081,143 @@ public function browseFrom($query, $params = null, $cursor = null)
...
+    {
+        return $this->client->request(
+            $this->context,
+            'PUT',
+            '/1/indexes/'.$this->urlIndexName.'/synonyms/'.urlencode($objectID).'?forwardToSlaves='.($forwardToSlaves ? 'true' : 'false')
+            .'&forwardToSlaves='.($forwardToSlaves ? 'true' : 'false'),
```
Comment (redox) [inline src/AlgoliaSearch/Index.php]: Double `forwardToSlaves`

**Comment by maxiloc** [comment]
Rebased :smile: @redox

**Comment by Nagriar** [comment]
👍

**Comment by redox** [comment]
LGTM 👍

**Pull request merged**